app/models/annotation/AnnotationService.scala

```diff
@@ -43,17 +39,20 @@ object AnnotationService
   with ProtoGeometryImplicits
   with LazyLogging {
 
-  private def selectSuitableTeam(user: User, dataSet: DataSet)(implicit ctx: DBAccessContext): Fox[ObjectId] = {
-    val selectedTeamOpt = dataSet.allowedTeams.intersect(user.teamIds).headOption
-    selectedTeamOpt match {
-      case Some(selectedTeam) => Fox.successful(ObjectId.fromBsonId(selectedTeam))
-      case None =>
-        for {
-          _ <- (user.isTeamManagerInOrg(user.organization) || user.isAdmin)
-          organization <- OrganizationSQLDAO.findOneByName(user.organization)
-          organizationTeamId <- OrganizationSQLDAO.findOrganizationTeamId(organization._id)
-        } yield organizationTeamId
-    }
+  private def selectSuitableTeam(user: UserSQL, dataSet: DataSet)(implicit ctx: DBAccessContext): Fox[ObjectId] = {
+    (for {
+      userTeamIds <- user.teamIds
+    } yield {
```

Comment:
What exactly is `flatten`ed here? Could we use [`async`](https://github.com/scala/scala-async) to circumvent those `(for {…} yield { for {} yield x}).flatten` patterns?